pkg/controllers/vdb/status_reconciler.go

```diff
@@ -172,7 +172,7 @@ func (s *StatusReconciler) calculateSubclusterStatus(ctx context.Context, sc *va
 		return err
 	}
 
-	for podIndex := int32(0); podIndex < int32(len(curStat.Detail)); podIndex++ {
+	for podIndex := int32(0); podIndex < int32(len(curStat.Detail)); podIndex++ { //nolint:gosec
```

Comment:
I see many occurrences of `//nolint:gosec`. Why is that?

Reply:
The new version of golangci-lint has a more strict type check. The type conversion int32(xxx) will be reported by go lint.